src/openrct2/ride/Ride.cpp

```diff
@@ -4833,91 +4833,139 @@ uint8_t ride_entry_get_vehicle_at_position(int32_t rideEntryIndex, int32_t numCa
+              SpriteGroupType::SlopeInverted, SpritePrecision::Sprites4 }, // TRACK_INLINE_TWIST_UNINVERTED
+            { SpriteGroupType::FlatBanked22, SpritePrecision::Sprites4, SpriteGroupType::FlatBanked45,
+              SpritePrecision::Sprites4, SpriteGroupType::InlineTwists, SpritePrecision::Sprites4,
+              SpriteGroupType::SlopeInverted, SpritePrecision::Sprites4 }, // TRACK_INLINE_TWIST_INVERTED
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60, SpritePrecision::Sprites4,
+              SpriteGroupType::Slopes75, SpritePrecision::Sprites4, SpriteGroupType::Slopes90,
+              SpritePrecision::Sprites4 }, // TRACK_QUARTER_LOOP_UNINVERTED
+            { SpriteGroupType::Slopes90, SpritePrecision::Sprites4, SpriteGroupType::SlopesLoop, SpritePrecision::Sprites4,
+              SpriteGroupType::SlopeInverted, SpritePrecision::Sprites4 }, // TRACK_QUARTER_LOOP_INVERTED
+            { SpriteGroupType::Slopes12, SpritePrecision::Sprites4 },      // TRACK_RAPIDS
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60, SpritePrecision::Sprites4,
+              SpriteGroupType::Slopes75, SpritePrecision::Sprites4, SpriteGroupType::Slopes90,
+              SpritePrecision::Sprites4 }, // TRACK_HALF_LOOP_UNINVERTED
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60, SpritePrecision::Sprites4,
+              SpriteGroupType::Slopes75, SpritePrecision::Sprites4, SpriteGroupType::Slopes90, SpritePrecision::Sprites4,
+              SpriteGroupType::SlopesLoop, SpritePrecision::Sprites4, SpriteGroupType::SlopeInverted,
+              SpritePrecision::Sprites4 },                             // TRACK_HALF_LOOP_INVERTED
+            { SpriteGroupType::SlopeFlat, SpritePrecision::Sprites4 }, // TRACK_WATERFALL
+            { SpriteGroupType::SlopeFlat, SpritePrecision::Sprites4 }, // TRACK_WHIRLPOOL
+            { SpriteGroupType::Slopes25, SpritePrecision::Sprites4, SpriteGroupType::Slopes60,
+              SpritePrecision::Sprites4 }, // TRACK_BRAKE_FOR_DROP
+            { SpriteGroupType::Corkscrews, SpritePrecision::Sprites4, SpriteGroupType::SlopeInverted,
+              SpritePrecision::Sprites4 }, // TRACK_CORKSCREW_UNINVERTED
+            { SpriteGroupType::Corkscrews, SpritePrecision::Sprites4, SpriteGroupType::SlopeInverted,
+              SpritePrecision::Sprites4 }, // TRACK_CORKSCREW_INVERTED
+            { SpriteGroupType::Slopes12, SpritePrecision::Sprites4, SpriteGroupType::Slopes25,
+              SpritePrecision::Sprites4 },                             // TRACK_HEARTLINE_TRANSFER
+            {},                                                        // TRACK_MINI_GOLF_HOLE
+            { SpriteGroupType::SlopeFlat, SpritePrecision::Sprites4 }, // TRACK_ROTATION_CONTROL_TOGGLE
+            { SpriteGroupType::Slopes60, SpritePrecision::Sprites4 },  // TRACK_SLOPE_STEEP_UP
+        };
```

Comment:
This is the source of the test failures. You can't change how this table works otherwise car rides are interpreted wrong.

Reply:
i tthink you just need to modify `RCT2RideTypeToOpenRCT2RideType` to use the new sprite precision stuff

Reply:
No actually this is all wrong. I can't find any vehicle that returns the correct supported pieces.

Reply:
See below